package.json

```diff
@@ -180,7 +180,7 @@
     "isomorphic-fetch": "^2.2.1",
     "jest": "^23.6.0",
     "jest-image-snapshot": "^2.7.0",
-    "jsdom": "^11.1.0",
+    "jsdom": "^15.2.1",
```

Comment:
This is what I was trying to accomplish with this PR. All other changes are to adapt to changes/problems introduced by this upgrade. Mostly just changing how certain document properties are mocked so that we can override and spy on them in tests.